website/source/docs/platform/k8s/upgrading.html.md

```diff
@@ -61,3 +61,81 @@ With the servers upgraded, it is time to upgrade the clients. To upgrade
+
+1. Run a Helm upgrade with the following config:
+   ```yaml
+   global:
+     tls:
+       enabled: true
+       # This configuration sets `verify_outgoing`, `verfiy_server_hostname`,
+       # and `verify_incoming` to `false` on servers and clients,
+       # which allows TLS-disabled nodes to join the cluster.
+       verify: false
+   server:
+     updatePartition: <number_of_server_replicas>
+   ```
+   This upgrade will trigger a rolling update of the clients, as well as any
+   other `consul-k8s` components, such as sync catalog or client snapshot deployments.
+1. Perform a rolling upgrade of the servers, as described in
+   [Upgrade Consul Servers](#upgrading-consul-servers).
+1. Repeat steps 1 and 2, turning on TLS verification by removing the
+   `global.tls.verify` property.
+
+### Gradual TLS Rollout with Consul Connect
+
+Because the Envoy proxy needs to talk to the Consul client agent regularly
+for service discovery, we can't enable TLS on the clients without also re-injecting
+TLS-enabled proxy into the application pods. To perform TLS rollout with minimal
+downtime, we recommend instead to add a new Kubernetes node pool and migrate your
+applications to it.
+
+1. Add a new identical node pool.
+1. Cordon all nodes in the old pool by running `kubectl cordon`
+   to ensure Kubernetes doesn't schedule any new workloads on those nodes.
+1. Create the following Helm config file for the upgrade:
+    ```yaml
+    global:
+      tls:
+        enabled: true
+        # This configuration sets `verify_outgoing`, `verfiy_server_hostname`,
+        # and `verify_incoming` to `false` on servers and clients,
+        # which allows TLS-disabled nodes to join the cluster.
+        verify: false
+    server:
+      updatePartition: <number_of_server_replicas>
+    client:
+      updateStrategy: |
+         type: OnDelete
+    ```
+   In this configuration, we're setting `server.updatePartition` to the number of
+   server replicas as described in [Upgrade Consul Servers](#upgrading-consul-servers)
+   and `client.updateStrategy` to `OnDelete` to manually trigger an upgrade of the clients.
+1. Run `helm upgrade` with the above config file. The upgrade will trigger an update of any
+   component except clients and servers, such as Consul Connect webhook deployment
+   or sync catalog deployment. Note that the sync catalog and the client
+   snapshot deployments will not be in the `ready` state until the clients on their
+   nodes are upgraded. It is OK to proceed to the next step without them being ready
+   because Kubernetes will keep the old deployment pod around, and so there will be no
+   downtime.
+1. Gradually perform an upgrade of the clients by deleting client pods on the new node
+   pool.
+1. Redeploy all your Connect-enabled applications. Now that the Connect injector is TLS-aware,
+   it will add TLS configuration to the sidecar proxy. Also, Kubernetes should schedule
+   these applications on the new node pool.
```

Comment:
Maybe add that one way to trigger a redeploy is to `kubectl drain` the nodes in the old pool.